lib/stdlib/doc/src/erl_tar.xml

```diff
@@ -125,24 +125,40 @@
     </list>
   </section>
 
+  <datatypes>
+    <datatype>
+      <name name="name_in_archive"/>
+    </datatype>
+    <datatype>
+      <name name="open_handle"/>
+    </datatype>
+    <datatype>
+      <name name="add_type"/>
+    </datatype>
+    <datatype>
+      <name name="tar_entry"/>
```

Comment:
This type and all the ones below are only used in `table/1,2`, so they can be moved there.